rts/Sim/Projectiles/WeaponProjectiles/WeaponProjectile.cpp

```diff
@@ -317,7 +317,7 @@ void CWeaponProjectile::UpdateGroundBounce()
 	if (!weaponDef->groundBounce && !weaponDef->waterBounce)
 		return;
 	// maximum number of bounce already reached?
-	if ((bounces + 1) > weaponDef->numBounce)
+	if (bounces == weaponDef->numBounce)
```

Comment:
Maybe some kind of `max(bounces, weaponDef->numBounce) == weaponDef->numBounce`?

Just to be safe in case bounces increased to more than numBounce (which was the idea for the original code I suppose).

Reply:
I'm fairly sure it is normally impossible for `bounces` to skip ahead of `weaponDef->numBounce` because:
* `bounces` only ever gets modified via the ++ operator below this check
* `numBounce` is taken from a weapon def which is static.

Reply:
Defensive coding is a good practice. A check against >= is safer than just ==. Even if it is fine now, code changes elsewhere in future may lead to problems.